earthaccess/auth.py

```diff
@@ -11,22 +12,35 @@
 
 from .daac import DAACS
 
+try:
+    user_agent = f"earthaccess v{importlib.metadata.version('earthacess')}"
```

Comment:
```suggestion
    user_agent = f"earthaccess v{earthaccess.__version__}"
```
Since we already do this work when earthaccess [is imported](https://github.com/nsidc/earthaccess/blob/main/earthaccess/__init__.py#L46), it'd probably make more sense to use the version reported by the package. 

If we need more error handling, I'd be inclined to move it to the `__init__.py`

Note: imports will also need to be adjusted.